### PR TITLE
Fix #1329 - Pack de Errors - Summon (PET)

### LINKED ISSUE #1250: Summons with targetdistance running away from target
Summons that are supposed to attack from distance (targetdistance=">0") are running away "infinitely" (until target is lost for their master), if attacked monster target them.
[Vid showing that problem.](https://gfycat.com/importantfastdesertpupfish)

To reproduce:
1. Summon creature that attack from distance (emberwing, minotaur archer etc)
2. Attack hostile monster
3. Let that monster target your summon instead of you
4. Your summon is running away like crazy

Expected behavior:
Your summon is walking one tile every 2 sec. [Video showing that](https://youtu.be/68AVCNMZ8Bs?t=490) - skullfrost is slowly walking away instead of running.

I tried to fix that myself for a while, but failed as my attempt didnt work.

Code to look at:
https://github.com/opentibiabr/otservbr-global/blob/9965cd24f4ce5b9851e6e95dfc2d7e1cbde3b574/src/creature.cpp#L895
https://github.com/opentibiabr/otservbr-global/blob/9965cd24f4ce5b9851e6e95dfc2d7e1cbde3b574/src/monster.cpp#L1361
https://github.com/opentibiabr/otservbr-global/blob/ea1d34d4a6b8a88a4eec6712c082a430eb4a4cde/src/monster.cpp#L805

### LINKED ISSUE #1329: Pack de Errors - Summon (PET)
1- Você apanha de  seu próprio summon.
image

2- summon indo parar dentro de água.
image

3- ao relogar, summon não desaparece.

4- todas magias de attack estão com effects / dano /  area, totalmente errados.

image

### PR DESCRIPTION
**Pull request by heberholanda**
1- Você apanha de seu próprio summon.
Fixed.

2- summon indo parar dentro de água.
Por enquanto vou ignorar isso.

3- ao relogar, summon não desaparece.
Esse é o comportamento esperado.

4- todas magias de attack estão com effects / dano / area, totalmente errados.
Fixed pelo @debone

### CODE REVIEW

File: data/spells/scripts/summon/summons.lua [hunk 1]
```diff
@@ -51,5 +51,6 @@ function onCastSpell(player, variant)
     player:setStorageValue(Storage.PetSummon, os.time() + 15*60) -- 15 minutes from now
     player:say("My Power your Power", TALKTYPE_MONSTER_SAY)
     addEvent(removePet, 15*60*1000, mySummon:getId()) --I think this isn't necessary
+    addEvent(removePet, 14*60*1000, player:sendTextMessage(MESSAGE_EVENT_ADVANCE, "Your summon will disappear in less than one minute."))
```
Comment (debone) [inline data/spells/scripts/summon/summons.lua]: Can we do this differently? It feels like running removePet function as a dummy function and expecting it to not do what it's supposed to do is a risky thing.

File: data/events/events.xml [hunk 2]
```diff
@@ -3,7 +3,7 @@
 	<!-- Creature Methods -->
 	<event class="Creature" method="onChangeOutfit" enabled="1" />
 	<event class="Creature" method="onAreaCombat" enabled="0" />
-	<event class="Creature" method="onTargetCombat" enabled="0" /><!-- removeCombatProtection on secure modes protection -->
+	<event class="Creature" method="onTargetCombat" enabled="1" /><!-- removeCombatProtection on secure modes protection -->
```
Comment (debone) [inline data/events/events.xml:6]: Looking at the history and what this target combat does, it'd be really good to know why it was turned off as there seems to be nothing wrong in having it on. Maybe @dudantas can give us a light here?

File: data/spells/scripts/summon/summons.lua (hunk 1 above)
Comment (heberholanda) [reply data/spells/scripts/summon/summons.lua]: my mistake, I updated.

File: data/spells/spells.xml [hunk 3]
```diff
@@ -1,16 +1,16 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <spells>
 	<!-- under construction summons -->
-	<instant group="support" spellid="194" name="Summon Skullfrost" words="utevo gran res eq" lvl="200" mana="1000"  cooldown="1800000" groupcooldown="2000" needlearn="0" script="summon/summons.lua">
+	<instant group="support" spellid="194" name="Summon Skullfrost" words="utevo gran res eq" lvl="200" mana="1000"  cooldown="18" groupcooldown="2000" needlearn="0" script="summon/summons.lua">
```
Comment (debone) [inline data/spells/spells.xml]: 18 ms? :P I was sure this was working?

File: data/spells/scripts/summon/summons.lua [hunk 4]
```diff
@@ -51,6 +51,6 @@ function onCastSpell(player, variant)
     player:setStorageValue(Storage.PetSummon, os.time() + 15*60) -- 15 minutes from now
     player:say("My Power your Power", TALKTYPE_MONSTER_SAY)
     addEvent(removePet, 15*60*1000, mySummon:getId()) --I think this isn't necessary
-    addEvent(removePet, 14*60*1000, player:sendTextMessage(MESSAGE_EVENT_ADVANCE, "Your summon will disappear in less than one minute."))
+    addEvent(function() player:sendTextMessage(MESSAGE_EVENT_ADVANCE, "Your summon will disappear in less than one minute.") end, 14*60*1000)
```
Comment (debone) [inline data/spells/scripts/summon/summons.lua:54]: Can't we have a "sendWarning" function? :)

**Review by dex-89 — APPROVED** [review]
Tested at emberwing. Passed. I can't attack my summon and summon can't attack me. But after logout and login summon disappears

File: data/events/events.xml (hunk 2 above)
Comment (dudantas) [reply data/events/events.xml:6]: This is to maintain retro pvp mode. It is the behavior of the buttons next to the equipments.
Taking damage from the summon itself is not a bug, it is part of retro-pvp.

**Pull request closed without merging**